### PR TITLE
Header & TOC tweaks

### PR DESCRIPTION
**Pull request by symbolpunk**
- incremental motion towards IA proposed [here](https://www.notion.so/Dev-Docs-Site-V2-Information-architecture-onboarding-and-next-steps-9ecd42550da2401ca71ee8dbb376aeb6) (internal only)

### CODE REVIEW

**Comment by TucksonDev** [comment]
If the menu entries change, should we change the title of those pages too?

**Comment by symbolpunk** [comment]
@TucksonDev Yep! I'm not sure if that'll be in this PR though :). If you'd like to help drive consistency, here's a fledgling convention & style guide -> [link](https://www.notion.so/arbitrum/Style-and-convention-guide-Docs-dafaf8d1ce34433d88c1ef3b4b99c19c))

meta -> flipping from `Draft` -> `Open` and assigning a reviewer is my signal for "ready for review" btw ~ definitely feel free to drop comments on `Drafts`, just wanted to share the way that I use various features when staging updates :)

**Comment by symbolpunk** [comment]
[keeping this scoped to header + sidebar, will break remaining consistency-driving work down into tasks]

File: website/sidebars.js [hunk 1]
```diff
@@ -21,20 +21,20 @@ const sidebars = {
     {
       type: "doc",
       id: "intro/intro",
-      label: "A Gentle Introduction"
+      label: "A gentle introduction: Arbitrum"
```
Comment (symbolpunk) [inline website/sidebars.js:24]: Eventual goal is to normalize the usage of Gentle Intros as the "informational onboarding" content type for each header section, while Quickstarts are offer "procedural onboarding".

- Details & next steps [here](https://www.notion.so/arbitrum/Dev-Docs-Site-V2-Information-architecture-onboarding-and-next-steps-9ecd42550da2401ca71ee8dbb376aeb6).
- Content types and conventions [here](https://www.notion.so/Style-and-convention-guide-Docs-dafaf8d1ce34433d88c1ef3b4b99c19c)

**Review by TucksonDev — APPROVED** [review]

**Pull request merged**